src/streams/register.rs

```diff
@@ -108,7 +112,11 @@ impl ReceiverRegister {
         if did_wrap {
             // The indices wrapped around the end of `packet_status`, we clear any entries between
             // the current sequence number and nack_check_from.
-            let start = (*seq - self.packet_index(*seq) as u64).into();
+
+            let start = (*self.nack_check_from - MISORDER_DELAY)
+                .max(seq.saturating_sub(self.packet_status.len() as u64 + 1))
+                .into();
```

Comment:
Can you talk me through this start value like I'm 5 years old.

Reply:
I addressed my other comment removing this